examples/hybrid/bubble3d_invariant_totalenergy.jl

```diff
@@ -0,0 +1,275 @@
+push!(LOAD_PATH, joinpath(@__DIR__, "..", ".."))
+
+using Test
+using StaticArrays, IntervalSets, LinearAlgebra, UnPack
+
+import ClimaCore:
+    ClimaCore,
+    slab,
+    Spaces,
+    Domains,
+    Meshes,
+    Geometry,
+    Topologies,
+    Spaces,
+    Fields,
+    Operators
+using ClimaCore.Geometry
+
+using Logging: global_logger
+using TerminalLoggers: TerminalLogger
+global_logger(TerminalLogger())
+
+function hvspace_3D(
+    xlim = (-π, π),
+    ylim = (-π, π),
+    zlim = (0, 4π),
+    xelem = 4,
+    yelem = 4,
+    zelem = 16,
+    npoly = 3,
+)
+    FT = Float64
+    vertdomain = Domains.IntervalDomain(
+        Geometry.ZPoint{FT}(zlim[1]),
+        Geometry.ZPoint{FT}(zlim[2]);
+        boundary_tags = (:bottom, :top),
+    )
+    vertmesh = Meshes.IntervalMesh(vertdomain, nelems = zelem)
+    vert_center_space = Spaces.CenterFiniteDifferenceSpace(vertmesh)
+
+    horzdomain = Domains.RectangleDomain(
+        Geometry.XPoint{FT}(xlim[1])..Geometry.XPoint{FT}(xlim[2]),
+        Geometry.YPoint{FT}(ylim[1])..Geometry.YPoint{FT}(ylim[2]),
+        x1periodic = true,
+        x2periodic = true,
+    )
+    horzmesh = Meshes.RectilinearMesh(horzdomain, xelem, yelem)
+    horztopology = Topologies.Topology2D(horzmesh)
+
+    quad = Spaces.Quadratures.GLL{npoly + 1}()
+    horzspace = Spaces.SpectralElementSpace2D(horztopology, quad)
+
+    hv_center_space =
+        Spaces.ExtrudedFiniteDifferenceSpace(horzspace, vert_center_space)
+    hv_face_space = Spaces.FaceExtrudedFiniteDifferenceSpace(hv_center_space)
+    return (hv_center_space, hv_face_space)
+end
+
+# set up 3D domain - doubly periodic box
+hv_center_space, hv_face_space = hvspace_3D((-500, 500), (-500, 500), (0, 1000))
+
+const MSLP = 1e5 # mean sea level pressure
+const grav = 9.8 # gravitational constant
+const R_d = 287.058 # R dry (gas constant / mol mass dry air)
+const γ = 1.4 # heat capacity ratio
+const C_p = R_d * γ / (γ - 1) # heat capacity at constant pressure
+const C_v = R_d / (γ - 1) # heat capacity at constant volume
+const T_0 = 273.16 # triple point temperature
+
+Φ(z) = grav * z
+
+function pressure(ρ, e, uvw, z)
```

Comment:
Need to change this. See [here](https://github.com/CliMA/ClimaCore.jl/blob/0707b365c9f2a539179e1f37524ba9a58ef72b85/examples/3dsphere/baroclinic_wave.jl#L134).